pkg/restore/restore.go

```diff
@@ -733,18 +733,59 @@ func (ctx *context) restoreResource(resource, namespace, resourcePath string) (a
 		createdObj, restoreErr := resourceClient.Create(obj)
 		if apierrors.IsAlreadyExists(restoreErr) {
 			equal := false
-			if fromCluster, err := resourceClient.Get(obj.GetName(), metav1.GetOptions{}); err == nil {
-				equal, err = objectsAreEqual(fromCluster, obj)
-				// Log any errors trying to check equality
+			fromCluster, err := resourceClient.Get(obj.GetName(), metav1.GetOptions{})
+			if err == nil {
+				// Remove insubstantial metadata
+				fromCluster, err := resetMetadataAndStatus(fromCluster)
 				if err != nil {
-					ctx.infof("error checking %s against cluster: %v", obj.GetName(), err)
+					ctx.infof("Error trying to reset metadata for %s: %v", kube.NamespaceAndName(obj), err)
+					addToResult(&warnings, namespace, err)
+					continue
 				}
+
+				// We know the cluster won't have the restore name label, so
+				// copy it over from the backup
+				restoreName := obj.GetLabels()[api.RestoreLabelKey]
+				addLabel(fromCluster, api.RestoreLabelKey, restoreName)
+
+				equal = equality.Semantic.DeepEqual(fromCluster, obj)
 			} else {
-				ctx.infof("Error retrieving cluster version of %s: %v", obj.GetName(), err)
+				ctx.infof("Error retrieving cluster version of %s: %v", kube.NamespaceAndName(obj), err)
```

Comment:
usually put the non-nil error case first, when then allows you to un-nest the non-error case, and move the declaration of `equal` inline, since your error case will be continuing.